share/completions/jj.fish

```diff
@@ -1 +1,10 @@
-jj util completion fish | source
+# The reason for `__this-command-does-not-exist` is that, if dynamic completion
+# is not implemented, we'd like to get an error reliably. However, the
+# behavior of `jj` without arguments depends on the value of a config, see
+# https://jj-vcs.github.io/jj/latest/config/#default-command
+if set -l completion (COMPLETE=fish jj __this-command-does-not-exist 2>/dev/null)
```

Comment:
`jj` instructions should probably state that it's only necessary for fish below version 4.
Else some users would load duplicate completion definitions (at least Git completions can be slow).
Could make it say something like `if [ -z "$(complete -C'jj ')" ]; COMPLETE=fish jj | source; end`
which does nothing if completions are already installed